cypress/integration/sGMR/voyage-report-form-validation.spec.js

```diff
@@ -0,0 +1,160 @@
+const faker = require('faker');
+const { getFutureDate, getPastDate } = require('../../support/utils');
+
+describe('Validate report form', () => {
+  let departureDateTime;
+  let departurePort;
+  let arrivalDateTime;
+  let arrivalPort;
+  let vessel;
+  let people;
+
+  before(() => {
+    cy.task('readFileMaybe', 'cypress/fixtures/users.json').then((data) => {
+      const user = JSON.parse(data);
+      cy.checkUserExists(user.email).then((userExist) => {
+        if (userExist === false) {
+          cy.registerUser();
+        }
+      });
+    });
+    cy.fixture('vessel.json').then((vesselObj) => {
+      vesselObj.regNumber = faker.random.number();
+      vesselObj.name = `${vesselObj.name}${faker.random.number()}`;
+      vessel = vesselObj;
+    });
+    cy.fixture('people.json').then((personObj) => {
+      personObj.documentNumber = faker.random.number();
+      personObj.firstName = `Auto-${faker.name.firstName()}`;
+      personObj.lastName = faker.name.lastName();
+      personObj.dateOfBirth = getPastDate(30, 'DD/MM/YYYY');
+      personObj.expiryDate = getFutureDate(3, 'DD/MM/YYYY');
+      people = personObj;
+    });
+
+    departurePort = 'Auto-Port of Hong Kong';
+    arrivalPort = 'Port of Felixstowe';
+    departureDateTime = getFutureDate(1, 'DD/MM/YYYY HH:MM');
+    arrivalDateTime = getFutureDate(2, 'DD/MM/YYYY HH:MM');
+  });
+
+  beforeEach(() => {
+    cy.fixture('users.json').then((user) => {
+      cy.login(user.email, user.password);
+    });
+    cy.navigation('Reports');
+    cy.url().should('include', '/reports');
+    cy.get('.govuk-tabs__list li')
+      .filter('.govuk-tabs__list-item--selected').find('p').should('contain', 'Draft');
+    cy.get('.govuk-button--start').should('have.text', 'Start now').click();
+    cy.url().should('include', '/save-voyage/page-1');
+  });
+
+  it('Should verify Departure details mandatory data', () => {
+    const errors = [
+      'You must enter a departure date',
+      'You must enter a departure time',
+      'You must enter a departure point',
+    ];
+    cy.saveAndContinue();
+
+    cy.get('.govuk-error-message').each((error, index) => {
+      cy.wrap(error).should('contain.text', errors[index]).and('be.visible');
+    });
+  });
+
+  it('Should verify Arraival details mandatory data', () => {
+    const errors = [
+      'You must enter an arrival date',
+      'You must enter an arrival time',
+      'You must enter an arrival point',
+    ];
+    cy.enterDepartureDetails(departureDateTime, departurePort);
+    cy.saveAndContinue();
+    cy.url().should('include', '/page-2');
+    cy.get('.govuk-button[type="button"]').click();
```

Comment:
I suggest using the `cy.contains` function as it's easier to read and doesn't depend on CSS classes, e.g.
```suggestion
    cy.contains('button', 'Continue').click();
```